### ISSUE TITLE
Roles & permissions widgets are not accessible

### ISSUE THREAD

**Issue by bradenmacdonald** [openedx/frontend-app-admin-console#48]
This UI does not seem to have been implemented with a11y in mind.

1. For a non-visual user / screenreader user, there is currently no way to distinguish included vs. excluded permissions on this view:
    <img width="696" height="278" alt="Image showing the problematic UI" src="https://github.com/user-attachments/assets/678b23d4-0067-4789-b66c-a47191fda2d5" />

2. The numbers after each role have no alt text associated with the "👤" icon, so a screen reader will just read it as e.g. "Library User 1" and it would be unclear what the number is for:
    <img width="228" height="74" alt="Image showing a confusing icon" src="https://github.com/user-attachments/assets/3c3df317-7a0b-4ebe-9666-47fbd9f9e077" />
    
    (The `<Icon src={Person} className="mr-1" />` in the `<CardTitle>` component is missing [the `screenReaderText` property](https://paragon-openedx.netlify.app/components/icon/#props-api) - "This value is recommended for use unless the Icon is being used in a way that is purely decorative or provides no additional context for screen reader users. This field should be thought of the same way an alt attribute would be used for image tags.")

3. On this page, the ✅ and ❌ icons are hidden entirely from screenreaders, so they'll just find a blank table:
     <img width="708" height="245" alt="Image highlighting the problematic area of the UI" src="https://github.com/user-attachments/assets/ed361243-00ba-4770-88ed-d6ba01b469bb" />

4. Likely other issues - these are just the ones I noticed so far.
